### DISCUSSION TITLE
RFC: Automerging dependabot PRs

### DISCUSSION THREAD

**Discussion by jamesgorrie** [guardian/dotcom-rendering#3910] [RFCs]
We would like to have automerging of @dependabot [PRs](https://github.com/guardian/dotcom-rendering/pulls/app%2Fdependabot).

When and which SemVer version type are considerations we should have when implementing this.

[Other teams are also doing this](https://github.com/guardian/source/pull/1253)

## Reading 

https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request

**Comment by jamesgorrie** [comment]
**SemVer type:** Major version automerging

**Reply by mxdvl** [reply]
I think it’s good to have visibility on those, so I would not want to have them merged automatically.

**Comment by jamesgorrie** [comment]
**SemVer type:** Minor / patch version automerging

**Reply by mxdvl** [reply]
There are a few packages that don’t strictly follow SemVer, such as `aws-cdk`. We need to filter these out.
Otherwise, I think this is a safe operation, provided all CI passes.

**Comment by jamesgorrie** [comment]
**When:** Core hours + an indication that there are people about e.g. a certain amount or a "Good morning" thread in team chat.

**Reply by mxdvl** [reply]
An alternative would be combining all dependabot PRs and then require a single manual click: https://github.com/hrvey/combine-prs-workflow

**Comment by jamesgorrie** [comment]
**When:** core hours with a buffer i.e. 11 - 16, Monday to Thursdays excluding holidays.

**Reply by mxdvl** [reply]
I think we should run at a specific time on specific days.

I would suggest Tue & Thu @ 11am & 3pm.

**Comment by DavidLawes** [comment]
In the [example](https://github.com/guardian/source/pull/1253) provided I notice auto-merging was being suggested for dev-dependencies only, is this something we could consider? Auto merging dev dependencies only would come with much less risk (?), so it could be a good starting point?

**Reply by mxdvl** [reply]
Sadly we no longer have this distinction between dependency types, since #4050

**Comment by SiAdcock** [comment]
Does Chromatic get triggered on dependabot PRs? I'd feel more confident if Chromatic could run on verbumps to visual libraries like React, Emotion and Source.

**Reply by mxdvl** [reply]
Yes, Chromatic runs all the time!
